pkg/virt-controller/watch/vmi.go

```diff
@@ -1305,6 +1298,33 @@ func (c *VMIController) virtlauncherAttachmentPods(virtlauncherPod *k8sv1.Pod) (
 	return attachmentPods, nil
 }
 
+func (c *VMIController) waitForFirstConsumerTemporaryPods(vmi *virtv1.VirtualMachineInstance, virtLauncherPod *k8sv1.Pod) ([]*k8sv1.Pod, error) {
+	var temporaryPods []*k8sv1.Pod
+
+	// Get all pods from the namespace
+	pods, err := c.listPodsFromNamespace(vmi.Namespace)
+	if err != nil {
+		return temporaryPods, err
+	}
+
+	for _, pod := range pods {
+		// Cleanup candidates are temporary pods that are either controlled by the VMI or the virt launcher pod
+		if !isTempPod(pod) {
+			continue
+		}
+
+		if controller.IsControlledBy(pod, vmi) {
+			temporaryPods = append(temporaryPods, pod)
+		}
+
+		if ownerRef := controller.GetControllerOf(pod); ownerRef != nil && ownerRef.UID == virtLauncherPod.UID {
```

Comment:
When we create a temporary pod, it always has VMI as a controller/owner, so why this additional check?

Reply:
and if this is not needed (is it?) then I would prefer to make the loop like this:
```
if isTempPod(pod) &&
   controller.IsControlledBy(pod, vmi) {
	temporaryPods = append(temporaryPods, pod)
}
```

Reply:
the hotplug temp trigger pods, they are owned by the virt-launcher
https://github.com/kubevirt/kubevirt/blob/main/pkg/virt-controller/services/template.go#L1354

Reply:
yes, right